File: src/main/java/onl/tesseract/event/IronTrapDoorEventHandler.java
package onl.tesseract.event;


import onl.tesseract.CreativePlayer;
import org.bukkit.Material;
import org.bukkit.block.Block;
import org.bukkit.block.data.type.TrapDoor;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerInteractEvent;
import org.bukkit.inventory.EquipmentSlot;

public class IronTrapDoorEventHandler implements Listener {
    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
    public void onClickTrapDoor(final PlayerInteractEvent event) {
        Block block = event.getClickedBlock();
        if (block == null || block.getType() != Material.IRON_TRAPDOOR || event.getHand() != EquipmentSlot.HAND)
            return;

        if (event.getPlayer().isSneaking()) {
            TrapDoor trapDoor = (TrapDoor) block.getBlockData();
            trapDoor.setOpen(!trapDoor.isOpen());
            block.setBlockData(trapDoor);
        }
    }
}

File: src/main/java/onl/tesseract/event/IronTrapDoorEventHandler.kt
package onl.tesseract.event


import org.bukkit.Material
import org.bukkit.block.data.type.TrapDoor
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerInteractEvent
import org.bukkit.inventory.EquipmentSlot

class IronTrapDoorEventHandler : Listener {
    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
    fun onClickTrapDoor(event: PlayerInteractEvent) {
        val block = event.clickedBlock
        if (block == null || block.type != Material.IRON_TRAPDOOR || event.hand != EquipmentSlot.HAND) return

        if (event.player.isSneaking) {
            val trapDoor = block.blockData as TrapDoor
            trapDoor.isOpen = !trapDoor.isOpen
            block.blockData = trapDoor
        }
    }
}